phoenix-core/src/it/java/org/apache/phoenix/end2end/index/ViewIndexIT.java

```diff
@@ -169,10 +171,67 @@ public void testDeleteViewIndexSequences() throws Exception {
         verifySequenceValue(null, sequenceName, sequenceSchemaName, Short.MIN_VALUE + 2);
         conn1.createStatement().execute("DROP VIEW " + fullViewName);
         conn1.createStatement().execute("DROP TABLE "+ fullTableName);
-        
+
         verifySequenceNotExists(null, sequenceName, sequenceSchemaName);
     }
-    
+
+    @Test
+    public void testDroppingColumnWhileCreatingIndex2() throws Exception {
+        String schemaName = "S1";
+        String tableName = generateUniqueName();
+        String viewSchemaName = "S1";
+        String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        String indexName = "IND_" + generateUniqueName();
+        String viewName = "VIEW_" + generateUniqueName();
+        String fullViewName = SchemaUtil.getTableName(viewSchemaName, viewName);
+
+        createBaseTable(schemaName, tableName, false, null, null, true);
+        Connection conn = getConnection();
+        conn.setAutoCommit(true);
+        conn.createStatement().execute("CREATE VIEW " + fullViewName + " AS SELECT * FROM " + fullTableName);
+        conn.commit();
+
+        final String[] failedMsg = new String[1];
+        final CountDownLatch doneSignal = new CountDownLatch(2);
+        Runnable r1 = new Runnable() {
+
+            @Override public void run() {
+                try {
+                    conn.createStatement().execute("CREATE INDEX " + indexName + " ON " + fullViewName + " (v1)");
+                } catch (SQLException e) {
+                    failedMsg[0] = e.getMessage();
```

Comment:
nit: Can we store and hence later assert the SQLException code rather than the exception message?